Add a GEMINI.md file with instructions for the CLI (#2895)
google/comprehensive-rust: STYLE.md

### Rust Code

When showing Rust code inline, please use the same spacing as `rustfmt`: `3 * x`
instead of `3*x`. However, feel free to remove newlines when it can make the

> code more compact and easier to understand, e.g., you can use

Reviewer: ```suggestion
code more compact and easier to understand, e.g., you can define a struct on one line if it is not the focus of your example:
```

<!-- dprint-ignore-end -->

> Please use this sparingly: enclose the code block in
> `<!-- dprint-ignore-start -->` and `<!-- dprint-ignore-end -->` to suppress the
> automatic formatting.

Reviewer: ```suggestion
Enclose the code block in
`<!-- dprint-ignore-start -->` and `<!-- dprint-ignore-end -->` to suppress the
automatic formatting. Please use this sparingly.
```

### Language and Tone

The courses are written in American English, so write "initialize", not
"initialise".

Use an informal, friendly, and concise tone. Remember that the courses are meant

> to be taught by experienced programmer to other experienced programmers. We

Reviewer: ```suggestion
to be taught by experienced an programmer to other experienced programmers. We
```

Author: I'll go with this instead:

```suggestion
to be taught by an experienced programmer to other experienced programmers. We
```

:smile:

> expect familiarity with programming, typically in a statically compiled language

Reviewer: Simply "compiled" or "statically typed"

like Java or C++. We don't explain common concepts known from that family of
languages, but we cannot assume familiarity with things like functional
programming.

## Speaker Notes

We have extended `mdbook` with support for speaker notes: content added between
`<details> ... </details>` tags is rendered in a special box that can be
collapsed or removed entirely from the slide.

- Unlike the main content, the speaker notes don't have to fit on a single
  slide.

- The speaker notes should expand on the topic of the slide. Use them to provide
  interesting background information for both the instructor and for students
  who look at the material outside of a class. Remember that many more people
  will read the course by themselves, so make the notes complete and useful even
  when there is no Rust expert around.

> - Avoid using speaker notes as a script for the instructor. When teaching the

Reviewer: ```suggestion
- Speaker notes are not a script for the instructor. When teaching the
```

>   course, instructors will only have a short time to glance at the notes.

Reviewer: ```suggestion
  course, instructors only have a short time to glance at the notes. Don't include full paragraphs for the instructor to read out loud.
```
